backend/core/followup_management_service.py

```
@@ -0,0 +1,278 @@
+        """
+        try:
+            with self.db_manager.get_connection() as conn:
+                cursor = conn.cursor()
+
+                # SECURITY FIX: Transaction handled automatically by context manager
+                # No manual BEGIN TRANSACTION needed with proper context usage
+
+                # Validate category exists and get question count
+                category = self.db_manager.get_followup_category(category_id)
+                if not category:
+                    raise ValueError(f"Category {category_id} not found")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)